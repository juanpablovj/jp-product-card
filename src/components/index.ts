import { ProductCard as ProductCardHOC } from './ProductCard';
import { ProductImage } from './ProductImage';
import { ProductTitle } from './ProductTitle';
import { ProductButtons } from './ProductButtons';
import { ProductCardHOCProps } from '../interfaces/interfaces';

const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
	Title: ProductTitle,
	Image: ProductImage,
	Buttons: ProductButtons,
});

export { ProductCard, ProductImage, ProductTitle, ProductButtons };
