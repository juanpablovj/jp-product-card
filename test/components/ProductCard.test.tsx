import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components';
import { product1 } from '../data/products';

const { act } = renderer;

describe('ProductCard', () => {
  test('should de mostrar el componente correctamente', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {() => <div>ProductCard</div>}
      </ProductCard>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should de incrementar el contador', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {({ count, increaseBy }) => (
          <>
            <h1>ProductCard</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(1)}></button>
          </>
        )}
      </ProductCard>
    );

    let tree = wrapper.toJSON();
    expect(wrapper).toMatchSnapshot();
    act(() => {
      (tree as any).children[2].props.onClick();
    });

    tree = wrapper.toJSON();

    expect((tree as any).children[1].children[0]).toBe('1');
  });
});
