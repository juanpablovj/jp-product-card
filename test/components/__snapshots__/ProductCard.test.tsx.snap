// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should de incrementar el contador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    ProductCard
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`ProductCard should de mostrar el componente correctamente 1`] = `
<div
  className="productCard undefined"
>
  <div>
    ProductCard
  </div>
</div>
`;
